### ISSUE TITLE
ASF projects can't bundle and ship jruby9k

### ISSUE THREAD

**Issue by madrob** [jruby/jruby#4603]
The `jruby-complete` jar bundles `racc/cparse-jruby.jar` which is labelled as LGPL licensed. This is a problem for ASF projects and means we won't be able to redistribute you guys.

@headius mentioned that what while the original racc native ext is LGPL, the racc runtime that is shipped with MRI falls under the Ruby license. The pieces that jruby uses are ported from the Ruby part, so this should just be a notational update.

To fix this we will need to:
1. Verify that the racc code we use is not LGPL!
2. Update headers on the relevant source files
3. Update `COPYING`

Also, would be good to file issues with tenderlove/racc for them to update their licensing to differentiate the non-LGPL bits, and another issue to release the differently licensed sources as separate gems.

**Comment by headius** [comment]
Unfortunately it's not simply a notational issue. I will clarify a few items:

* The racc runtime, which includes the C and Java extensions, is licensed under the same terms as Ruby (MIT or BSD-ish).
* The racc compiler is licensed as LGPL.
* When we started shipping the entire racc gem, we inadvertently pulled in the LGPL portions.

I've filed a few PRs and issues with racc to get their licensing straightened up, including splitting the non-LGPL runtime off into a separate gem. I also updated the Java ext to point out that it's Ruby-licensed, but this is purely notational; it was always licensed this way.

On the JRuby side, I'll push a commit that commits the non-LGPL pieces directly to our repository. This is a temporary situation until tenderlove/racc#85 happens and we can again source our racc logic from a gem. This unfortunately means committing a (very small) binary for the parser and updating COPYING to point out where it came from.
